### PR TITLE
Add docs for decorator before export in babel generator

### LINKED ISSUE #2405: [babel.generate]:  generate convert ast to code
## Bug Report

- [ ] I would like to work on a fix!

**Current behavior**

-  [Codesandbox](https://codesandbox.io/s/clever-pascal-yd3l2)

first, I use babel  parse a angular module which decorators before export

```js
const preCode = `
    @NgModule({})
    export class AppModule {};
  `;
  const ast = parse(preCode, {
    sourceType: "module",
    plugins: [
      "decorators-legacy" // decoratorsBeforeExport: true
    ]
  });
```

then, I use generate to convert ast to code.

```js
const code = generate(ast).code;
// export @NgModule({})
// class AppModule {}
```
but code by generate is export before decorators

**Expected behavior**

I expect generate could support a option like` decoratorsBeforeExport` or ` decorators-legacy`

**Environment**

- Babel version(s): [7+]
- Node/npm version: [ Node 10+/npm 6+]
- OS: [Mac OS]

**Possible Solution**

I hope someone can tell me how to do it。

### PR DESCRIPTION
**Pull request by AjayPoshak**
Fixes #2405

### CODE REVIEW

File: docs/generator.md [hunk 1]
```diff
@@ -46,6 +46,8 @@ Options for formatting output:
 | filename               | string              |                 | Used in warning messages                                                                                                                                                                                                                                           |
 | jsonCompatibleStrings  | boolean             | `false`         | Set to true to run `jsesc` with "json": true to print "\u00A9" vs. "©";                                                                                                                                                                                            |
 | jsescOption            | object              |                 | Use `jsesc` to process literals. `jsesc` is applied to numbers only if `jsescOption.numbers` (added in `v7.9.0`) is present. You can customize `jsesc` by [passing options](https://github.com/mathiasbynens/jsesc#api) to it.                                     |
+| decoratorsBeforeExport | boolean              |                 | Set to `true` to enable decorators before export in the generated code.                            |
```
Comment (nicolo-ribaudo) [inline docs/generator.md:49]: Nit:

```suggestion
| decoratorsBeforeExport | boolean              |                 | Set to `true` to print decorators before `export` in the generated code.                            |
```

Comment (JLHwung) [reply docs/generator.md:49]: We use `output` when mentioning generated code.
```suggestion
| decoratorsBeforeExport | boolean              |                 | Set to `true` to print decorators before `export` in output
```

**Comment by existentialism** [comment]
Not required for this PR, but I think the list of options should be sorted alphabetically.

**Comment by existentialism** [comment]
@AjayPoshak thanks

**Pull request merged**